transport/graphsync/graphsync.go

```diff
@@ -126,29 +126,24 @@ func (t *Transport) OpenChannel(ctx context.Context,
 			Data: bz}
 		exts = append(exts, doNotSendExt)
 	}
-	_, errChan := t.gs.Request(internalCtx, dataSender, root, stor, exts...)
+	responseChan, errChan := t.gs.Request(internalCtx, dataSender, root, stor, exts...)
 
-	go t.executeGsRequest(ctx, channelID, errChan)
+	go t.executeGsRequest(ctx, channelID, responseChan, errChan)
 	return nil
 }
 
-func (t *Transport) consumeResponses(ctx context.Context, errChan <-chan error) error {
+func (t *Transport) consumeResponses(responseChan <-chan graphsync.ResponseProgress, errChan <-chan error) error {
 	var lastError error
-	for {
-		select {
-		case <-ctx.Done():
-			return errContextCancelled
-		case err, ok := <-errChan:
-			if !ok {
-				return lastError
-			}
-			lastError = err
-		}
+	for range responseChan {
+	}
```

Comment:
@hannahhoward Don't we need to do anything with the responses here ? Who are these responses intended for ?

Reply:
It's another way to read response data -- in direct IPLD node format. They are still saved to the blockstore anyway so we don't need to do anything with them